src/finch/README.md

```diff
@@ -0,0 +1,45 @@
+# Finch Tensor: Developer README
+
+> **Note:** This README is AI-generated and may contain errors or omissions. If you find any inaccuracies or improvements, please update this file for the benefit of all contributors.
```

Comment:
Please read the readme yourself and correct any AI errors. Although AI is a helpful coding tool, I'd appreciate if everyone would personally stand by the code they are asking others to review. Perhaps you could add a statement to this effect in the readme too.